bundles/AdminBundle/Controller/Admin/DataObject/ClassController.php

```diff
@@ -39,7 +39,7 @@
  *
  * @internal
  */
-class ClassController extends AdminController implements KernelControllerEventInterface
+class ClassController extends AdminAbstractController implements KernelControllerEventInterface
```

Comment:
How do we deal with the `trans()` method here?

Reply:
We keep it in v10 for compatibility reasons, but we take it away in v11, because actions requiring the translator should use DI to get it.

Reply:
Just to be more clear, we keep it in the old deprecated `AdminController` and doesn't implement that in new abstract controller `AdminAbstractController` so the controllers already extending the new abstract should implement on their on own.

Reply:
@dvesh3 yes 👍